### DISCUSSION TITLE
Is there a way to match a pattern an exact number of times ?

### DISCUSSION THREAD

**Discussion by loloof64** [petitparser/dart-petitparser#133] [Q&A] [answered]
Hi ! Is there a way to match a pattern a given number of times ?

For example, I want to match the pattern `pattern('dfr')` 4 times, is there a simple way to write this ?
Or even for a range : `pattern('dfr')` between 2 and 4 times ?

Otherwise, if I must write it several times, is there a way to catch the group as "a single match" ? I mean, it is grouped as if I wrote it in another rule.

Regards

**Comment by renggli** [comment, marked as answer]
There is [times(count)](https://pub.dev/documentation/petitparser/latest/parser/PossessiveRepeatingParserExtension/times.html) for an exact number of repetitions, and [repeat(min, max)](https://pub.dev/documentation/petitparser/latest/parser/PossessiveRepeatingParserExtension/repeat.html) for a range.

```dart
final exact = pattern('dfr').times(4);
final range = pattern('dfr').range(2, 4);
```

**Reply by loloof64** [reply]
Thank you very much 😃 

Exactly what I was looking for 😃
